archaius2-core/src/main/java/com/netflix/archaius/DefaultPropertyFactory.java

```diff
@@ -130,7 +131,7 @@ public <T> Property<T> asType(Function<String, T> mapper, String defaultValue) {
                         try {
                             return mapper.apply(value);
                         } catch (Exception e) {
-                            LOG.warn("Invalid value '{}' for property '{}'", propName, value);
+                            LOG.error("Invalid value '{}' for property '{}'. Will return the default instead.", propName, value);
```

Comment:
I think due to the implementation of the above asType method, this just ends up returning null (this is what we observed in https://netflix.slack.com/archives/C141EJAJU/p1723751704356699)

Reply:
Yeah, I saw that. It's the same as in the `get()` method. I'll delve into fixing those two and any others I find in a separate PR, I want that one to be as clean as I can so it can be called out in the release notes and also it's easy to pinpoint if/when people complain about a behavior change.

Reply:
OK LGTM if this is assuming we are changing to that behavior, though this error message is technically a little off until we get to that point